### PR TITLE
Fixed sort article discussion posts by Author, Replies or Last Post

### PR DESCRIPTION
**Pull request by gauravghati**
@LeoMcA or @akatsoulas Can you look into this PR ...?

There was an extra line, which was causing this error, now it's working.
This PR fixes [issue#529](https://github.com/mozilla/sumo-project/issues/529)
sort

### CODE REVIEW

**Comment by LeoMcA** [comment]
Nearly there, but this is causing a test to fail (run it locally with `DEBUG=False ./manage.py test kitsune/kbforums/tests/test_templates.py:ThreadsTemplateTests.test_locale_discussions_ignores_sticky`):

```
======================================================================
FAIL: kitsune/kbforums/tests/test_templates.py:ThreadsTemplateTests.test_locale_discussions_ignores_sticky
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/app/kitsune/kbforums/tests/test_templates.py", line 341, in test_locale_discussions_ignores_sticky
    assert title.startswith("A thread with a very very long")
AssertionError

----------------------------------------------------------------------
```

I think we still need this `order_by("-last_post__created")` but as the default in `sort_threads()` rather than overwriting the ordering that function applies.

**Comment by gauravghati** [comment]
Done! all tests are passing locally now, I'm kinda new with CI-testing. :sweat_smile:

**Comment by LeoMcA** [comment]
No worries! And apologies, I led you down slightly the wrong path here:

It's true that in the `locale_discussions` view, we want `sort_threads` to use `order_by("-last_post__created")` if nothing has been selected, but in the `threads` view (on [L68](https://github.com/mozilla/kitsune/blob/85445d0fcdb62f124a20b21beefb31385da69b24/kitsune/kbforums/views.py#L68)) we still want it to use the default order which puts "sticky" threads at the top (by calling `all()`).

Also the `locale_discussions` view now has the slightly odd behaviour that clicking on "Last Post" the first time does nothing at all (because it's already sorting most recent to oldest).

I reckon both of the above can be solved by going with your original approach of just removing the line entirely, and modifying the `except ValueError` values of `sort` and `desc`.

File: kitsune/kbforums/views.py [hunk 1]
```diff
@@ -446,7 +446,6 @@ def locale_discussions(request):
     threads_ = sort_threads(threads, sort, desc)
 
     # Ignore sticky-ness:
```
Comment (LeoMcA) [inline kitsune/kbforums/views.py:448]: This comment refers to the line which was removed, so should be removed too

**Comment by gauravghati** [comment]
> It's true that in the `locale_discussions` view, we want `sort_threads` to use `order_by("-last_post__created")` if nothing has been selected, but in the `threads` view (on [L68](https://github.com/mozilla/kitsune/blob/85445d0fcdb62f124a20b21beefb31385da69b24/kitsune/kbforums/views.py#L68)) we still want it to use the default order which puts "sticky" threads at the top (by calling `all()`).

Now defaults for `thread` view are sort=0 and desc=0 which will call `all()`, and for `locale_discussions` defaults are sort=5 and desc=1 which will call `order_by("-last_post__created")`.

> Also the `locale_discussions` view now has the slightly odd behaviour that clicking on "Last Post" the first time does nothing at all (because it's already sorting most recent to oldest).
 
This also get resolved by changing defaults to `sort=5 and desc=1`. 
Please review and let me know if there are any other changes.

File: kitsune/kbforums/views.py (hunk 1 above)
Comment (gauravghati) [reply kitsune/kbforums/views.py]: Removed!

File: kitsune/kbforums/views.py [hunk 2]
```diff
@@ -433,20 +433,17 @@ def locale_discussions(request):
         document__locale=request.LANGUAGE_CODE, document__allow_discussion=True
     )
     try:
```
Comment (akatsoulas) [inline kitsune/kbforums/views.py:435]: Do we need the `try/except` clause here at all?

Comment (LeoMcA) [reply kitsune/kbforums/views.py:435]: I guess it's in case one of the arguments isn't castable to an int (like `&desc=foobar`)

**Review by LeoMcA — APPROVED** [review]
Great! :rocket:

**Pull request merged**